Raspberry Pi/RaspberrySide/VisionManager.py

```diff
@@ -83,18 +84,28 @@ def updateImage(self):
         if self.isObjectDetected:
             self.currentImageObject.didUpdateVar = False	
 
-        didGetImage,frame = self.cam.read()
+        didGetImage, new_frame = self.cam.read()
+        if not didGetImage:  # Can't event grab one frame - might be a problem
+            logger.error("Couldn't Read Image from self.cam!")
+            return
+        
+        for _ in xrange(self.READ_BUFFER_AMOUNT - 1):  # minus one, since we already grabbed one frame.
+            frame = new_frame
+            didGetImage, new_frame = self.cam.read()  # Grab another frame
+            if not didGetImage:
+                break  # Seems like the buffer is empty - lets use the last frame
+            
+        else:  # this occours if the loop wasn't broken - lets use the new frame
+            frame = new_frame
+            
         # maybe resize changes edges of u? causing the u to be smaller of bigger
         frame = cv2.resize(frame, (resizedImageWidth,resizedImageHeight))
         frame = self.rotateImage(frame)
-        if didGetImage:
-            if self.currentImage == None and self.imageHeight == None and self.imageWidth == None:
-                self.currentImage = frame
-                self.__setImageScales()
-            else:
-                self.currentImage = frame
-        else:
-            logger.error("Couldn't Read Image from self.cam!")
+    
+        if self.currentImage == None and self.imageHeight == None and self.imageWidth == None:
```

Comment:
Cleared up this if-statement along with my changes - make sure it still makes sense to you.